ohara-kafka/src/main/java/com/island/ohara/kafka/connector/csv/source/CsvDataReader.java

```
@@ -0,0 +1,110 @@
+ * limitations under the License.
+ */
+
+package com.island.ohara.kafka.connector.csv.source;
+
+import com.island.ohara.common.util.CommonUtils;
+import com.island.ohara.kafka.connector.RowSourceContext;
+import com.island.ohara.kafka.connector.RowSourceRecord;
+import com.island.ohara.kafka.connector.storage.Storage;
+import java.io.*;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CsvDataReader implements DataReader {
+  private static final Logger LOG = LoggerFactory.getLogger(CsvDataReader.class);
+
+  private final Storage storage;
+  private final CsvSourceConfig config;
+  private final RowSourceContext context;
+  private final CsvOffsetCache offsetCache;
+
+  public static CsvDataReader of(
+      Storage storage, CsvSourceConfig config, RowSourceContext context) {
+    return new CsvDataReader(storage, config, context);
+  }
+
+  public CsvDataReader(Storage storage, CsvSourceConfig config, RowSourceContext context) {
+    this.storage = storage;
+    this.config = config;
+    this.context = context;
+    this.offsetCache = new CsvOffsetCache();
+  }
+
+  public List<RowSourceRecord> read(Path file) {
+    try {
+      offsetCache.update(context, file.toString());
+      CsvRecordConverter converter =
+          new CsvRecordConverter.Builder()
+              .path(file.toString())
+              .topics(config.topics())
+              .offsetCache(offsetCache)
+              .schema(config.schema())
+              .build();
+
+      List<RowSourceRecord> records;
+      try (BufferedReader reader =
+          new BufferedReader(
+              new InputStreamReader(
+                  storage.open(file.toString()), Charset.forName(config.encode())))) {
+        records = converter.convert(reader.lines());
+      }
+
+      handleCompletedFile(file.toString());
```

should we swallow the exception for completed files? the exception of handling error files have been swallowed.

---

address it